src/hotspot/share/opto/addnode.cpp

```diff
@@ -216,22 +216,19 @@ Node *AddNode::Ideal(PhaseGVN *phase, bool can_reshape) {
 // the other input's symbols.
 const Type* AddNode::Value(PhaseGVN* phase) const {
   // Either input is TOP ==> the result is TOP
-  const Type *t1 = phase->type( in(1) );
-  const Type *t2 = phase->type( in(2) );
-  if( t1 == Type::TOP ) return Type::TOP;
-  if( t2 == Type::TOP ) return Type::TOP;
-
-  // Either input is BOTTOM ==> the result is the local BOTTOM
-  const Type *bot = bottom_type();
-  if( (t1 == bot) || (t2 == bot) ||
-      (t1 == Type::BOTTOM) || (t2 == Type::BOTTOM) )
```

Comment:
We can  keep (t1 == Type::BOTTOM || t2 == Type::BOTTOM) condition intact.

Reply:
I don't think `Type::BOTTOM` can appear here, there are other nodes (such as convert nodes) that do not check for `Type::BOTTOM`. Maybe someone could shed light on this, please?

Reply:
I don't think `Type::BOTTOM` is possible here either.

Reply:
The JVM is strongly typed so I don't think anything that is not higher than the local bottom can appear here. Note that the nodes will always return the local bottom or higher on `Value` invocation.